istio/src/main/scala/io/buoyant/k8s/istio/IstioDataFlow.scala

```diff
@@ -0,0 +1,71 @@
+package io.buoyant.k8s.istio
+
+import com.twitter.finagle.Path
+import io.buoyant.router.context.DstBoundCtx
+
+/**
+ * An abstraction applicable to both Requests and Responses in an Istio service mesh.
+ *
+ * In an Istio setup, both requests and responses are augmented with [[IstioAttribute]]s that can
+ * be used in various ways. While some attributes are specific to either requests or responses, others
+ * are common in both types of objects.
+ *
+ */
+trait IstioDataFlow {
```

Comment:
All of the information this trait provides is derived from the DstBoundCtx and not from any underlying request or response object.  So it seems a bit misleading to attach this to IstioRequest and IstioResponse objects.  It is also confusing that the `vals` on this trait are computed from the DstBoundCtx and saved at constructor time.  This means that the exact position in the stack where this object is constructed will determine this values, which is pretty non-intuitive.  

I would try to keep IstioDataFlow separate from the Request/Response objects and try to make it clear that constructing an IstioDataFlow takes a snapshot of the current DstBoundCtx.

Reply:
You are absolutely right, and I had felt some of this already. Unfortunately this branch didn't pick the changes, I've added them: https://github.com/linkerd/linkerd/pull/1650/files#diff-042aed18959e3a7caabf44f4294034ef

I am still not happy with `CurrentIstioPath`, but I ran out of ideas here.